linux/utils/install_uninstall_package.yml

```diff
@@ -116,7 +116,21 @@
             allow_downgrade: true
             state: "{{ package_state }}"
           delegate_to: "{{ vm_guest_ip }}"
-          when: guest_os_ansible_pkg_mgr | lower == "dnf"
+          when:
+            - guest_os_ansible_pkg_mgr | lower == "dnf"
+            - guest_os_ansible_distribution != "Kylin Linux Advanced Server"
+
+        - name: "{{ package_manage_op }} packages on {{ guest_os_ansible_distribution }} {{ guest_os_ansible_distribution_ver }}"
+          ansible.builtin.dnf:
+            name: "{{ package_manage_list }}"
+            allow_downgrade: true
+            state: "{{ package_state }}"
+            use_backend: "dnf4"
+          delegate_to: "{{ vm_guest_ip }}"
+          when:
+            - guest_os_ansible_pkg_mgr | lower == "dnf"
+            - guest_os_ansible_distribution == "Kylin Linux Advanced Server"
+            - guest_os_ansible_distribution_major_ver | int >= 10
```

Comment:
You don't need to add a separate task for Kylin Server V10. You just need to insert below line after line 117:
```
    use_backend: "{{ 'dnf4' if guest_os_ansible_distribution == "Kylin Linux Advanced Server" else 'auto' }}"
```

Or, to be more specific, you can get dnf major version by `dnf --version`. If it's 4.x, the backend is `dnf4`; if it's 5.x, the backend is `dnf5`.